docs/stock/data-grouping.md

```diff
@@ -1,24 +1,111 @@
+
+##### [Approximation](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.approximation)
+With the `approximation` option, you can decide how the final value for each group should be calculated. By default it depends on the series type: 
 *   Line type series use an _average_ approximation.
 *   Column type series compute the _sum_.
 *   Range type series compute the union's _range_.
 *   OHLC type series compute the union's open, high, low and close (OHLC) values.
 
-Grouping is activated when there are many data points in the chart. As well as increasing performance it makes it easier to spot trends in a chart.
 
-Data grouping is a Highcharts Stock feature and is enabled by default. To see dataGrouping options see the [API reference](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping).
+but you can use the custom function to calculate the final value for each group. Here is an example of custom approximation function:
+
+<iframe width="600" height="450" src="https://www.highcharts.com/samples/embed/stock/plotoptions/series-datagrouping-approximation" allowfullscreen></iframe>
+
+##### [Units](https://api.highcharts.com/highstock/series.line.dataGrouping.units)
+The available data grouping units can be set using `units`. By default, it's an array of time units (millisecond, second, minute, hour, day, week, month, year) with their allowed multiples.
+
+
+In the example below, the data grouping is set to only group data points into one, two or three days.
+```js
+Highcharts.stockChart('container', {
+    series: [{
+        dataGrouping:{
+            units: [['day', [1, 2, 3]]]
+        },
+        data: [...]
+    }]
+});
+```
+
+##### [groupAll](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.groupAll)
+
+Default behavior only groups the points within the visible range of the chart, but sometimes it's useful to group all the points. This can be achieved by setting `groupAll` to `true`.
+
+```js
+Highcharts.stockChart('container', {
+  series: [{
+    dataGrouping:{
+      groupAll: true
+    },
+    data: [...]
+  }]
+});
+```
+
+The points that would be outside of the visible range might have an impact on the values of the visible groups.
+Here is an example of a chart where you can see the difference between the two options:
+
+<iframe width="600" height="450" src="https://www.highcharts.com/samples/embed/stock/plotoptions/series-datagrouping-groupall" allowfullscreen></iframe>
+
+
+##### [forced](https://api.highcharts.com/highstock/plotOptions.series.dataGrouping.forced)
+
+Sometimes it's useful to force the data grouping to kick in, event if all of the data points can be easily fitted into the visible range. This can be achieved by setting `forced` to `true`, while also deterimining the `units` option. This way, the series will choose the lowest available option to group the points. This can be used to smoothen the series, similar to what indicators like SMA do.
```

Comment:
```suggestion
Sometimes it's useful to force the data grouping to kick in, even if all of the data points can be easily fitted into the visible range. This can be achieved by setting `forced` to `true`, while also deterimining the `units` option. This way, the series will choose the lowest available option to group the points. This can be used to smoothen the series, similar to what indicators like SMA do.
```